src/test/resources/features/EditShelter.feature

```diff
@@ -0,0 +1,51 @@
+Feature: Edit Shelter
+  In order to update shelter information
+  As a shelter manager
+  I want to edit a shelter
+
+  Background:
+    Given There is a registered user with username "user" and password "existing" and email "user@sample.app"
+    Given There is a registered admin with username "admin" and password "admin" and email "admin@smaple.app"
+    Given There is a registered volunteer with username "volunteer" and password "volunteer"
+    And There is already a shelter with name "Shelter 1" email "shelter@sample.com" and mobile "999999999"
+
+  Scenario: Edit shelter without being logged in
+    Given I'm not logged in
+    When I update the shelter with name "Shelter 1" to name "Another Shelter" email "shelter@sample.app" and mobile "123123123"
+    Then The response code is 401
+    And The error message is "Unauthorized"
+
+  Scenario: Edit shelter with user
+    Given I login as "client" with password "existing"
+    When I update the shelter with name "Shelter 1" to name "Another Shelter" email "shelter@sample.app" and mobile "123123123"
+    Then The response code is 401
+    And The error message is "Unauthorized"
+
+  Scenario: Edit shelter with admin
+    Given I login as "admin" with password "admin"
+    When I update the shelter with name "Shelter 1" to name "Another Shelter" email "shelter@sample.app" and mobile "123123123"
+    Then The response code is 200
+
+  Scenario: Edit shelter with volunteer
+    Given I login as "volunteer" with password "volunteer"
+    When I update the shelter with name "Shelter 1" to name "Another Shelter" email "shelter@sample.app" and mobile "123123123"
+    Then The response code is 200
```

Comment:
You should also check that the update has been materialised in the backend. For that you can add an additional step where a GET is performed using the shelter ID and the you check in the returned JSON, using JSONPath, that the updated name is returned now

Reply:
Just pushed the implementation of the specified step and changes in the .feature file